core/templates/pages/learner-dashboard-page/community-lessons-tab.component.spec.ts

```diff
@@ -779,4 +804,38 @@ describe('Community lessons tab Component', () => {
 
     expect(modalSpy).toHaveBeenCalled();
   }));
+
+  it('should return true when there are no lessons', () => {
+    expect(component.isLearnerStateEmpty()).toBeTrue();
+  });
+
+  it('should return false when there are in-progress lessons', () => {
+    component.totalIncompleteLessonsList = [
+      LearnerExplorationSummary.createFromBackendDict(sampleExploration),
+    ];
+    fixture.detectChanges();
+
+    expect(component.isLearnerStateEmpty()).toBeFalse();
+  });
+
+  it('should return false when there are completed lessons', () => {
+    component.totalCompletedLessonsList = [
+      LearnerExplorationSummary.createFromBackendDict(sampleExploration),
+    ];
+    fixture.detectChanges();
+
+    expect(component.isLearnerStateEmpty()).toBeFalse();
+  });
+
+  it('should return false when there are completed and in-progress lessons', () => {
+    component.totalCompletedLessonsList = [
+      LearnerExplorationSummary.createFromBackendDict(sampleExploration),
```

Comment:
I noticed that we are repeatedly creating the summary from the backend dict, how about creating it once where you define the sampleExploration and reusing it in tests?

Reply:
Done.